debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/RecordMakers.java

```diff
@@ -87,22 +218,17 @@ public RecordsForCollection forCollection(CollectionId collectionId) {
         private final SourceInfo source;
         private final Map<String, ?> sourcePartition;
         private final String topicName;
-        private final Schema keySchema;
         private final Schema valueSchema;
         private final Function<Document, String> valueTransformer;
         private final BlockingConsumer<SourceRecord> recorder;
 
         protected RecordsForCollection(CollectionId collectionId, SourceInfo source, String topicName, AvroValidator validator,
-                Function<Document, String> valueTransformer, BlockingConsumer<SourceRecord> recorder) {
+                                       Function<Document, String> valueTransformer, BlockingConsumer<SourceRecord> recorder) {
             this.sourcePartition = source.partition(collectionId.replicaSetName());
             this.collectionId = collectionId;
             this.replicaSetName = this.collectionId.replicaSetName();
             this.source = source;
             this.topicName = topicName;
-            this.keySchema = SchemaBuilder.struct()
```

Comment:
@gunnarmorling @rhauch this was removed by me but simply because when trying to do it type specific there is then not a single fixed pre-defined key schema anymore. instead the key schema is taken from the key struct the get's build based on what we would need. see e.g. the createRecords method https://github.com/debezium/debezium/pull/258/files/43ad10a498b4c3c7249d68071a9eec3c9547f704#diff-13c987a1e79d5f224c7f90145b80b979R319